elytron/src/main/java/org/wildfly/extension/elytron/AuthenticationClientDefinitions.java

```diff
@@ -328,7 +334,16 @@ static ResourceDefinition getAuthenticationClientDefinition() {
                 }
 
                 final Function<AuthenticationConfiguration, AuthenticationConfiguration> finalConfiguration = configuration;
-                return () -> finalConfiguration.apply(null);
+                return () -> {
+                    try {
+                        return finalConfiguration.apply(null);
+                    } catch (IllegalStateException e) {
+                        if (e.getCause() != null) {
+                            throw new StartException(e.getCause().getMessage());
+                        }
+                        throw new StartException(e.getMessage());
```

Comment:
I think both of these throws would be better if they included a cause so the stack trace is more useful.